test/continuous-test-suite-mcp-infra.ts

```
@@ -0,0 +1,1228 @@
+  }
+}
+
+async function testWiredMiddleware(): Promise<void> {
+  logSection("Wired Middleware Integration");
+
+  // Test 1: Middleware executes on tool call
+  try {
+    const sdk = new NeuroLink();
+    const middlewareCalls: string[] = [];
+
+    sdk.useToolMiddleware(async (_tool, _params, _context, next) => {
```

Biome (2.4.15)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

(lint/correctness/useHookAtTopLevel)